### PR TITLE
server, test: Refactor NodeCluster and ServerCluster

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: ref #14401

What's Changed:

```commit-message
Parameterize NodeCluster and ServerCluster
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by SpadeA-Tang — APPROVED** [review]

**Review by SpadeA-Tang — APPROVED** [review]

File: components/server/src/lib.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@ extern crate tikv_util;
 
 #[macro_use]
 pub mod setup;
-// This mod is exported to make convenience for creating TiKV-like servers.
+/// This mod is exported to make convenience for creating TiKV-like servers.
```
Comment (tabokie) [inline components/server/src/lib.rs:12]: Put it at the top of comment.rs, like `//! xxx`.

**Review by tabokie — APPROVED** [review]

**Pull request merged**